Sources/PackageGraph/Pubgrub.swift

```diff
@@ -163,13 +164,46 @@ struct Term<Identifier: PackageContainerIdentifier>: Equatable, Hashable {
     /// - The package version has to match all assignments.
     func isValidDecision(for solution: PartialSolution<Identifier>) -> Bool {
         for assignment in solution.assignments where assignment.term.package == package {
-            guard !assignment.isDecision else { return false }
-            guard satisfies(other: assignment.term) else { return false }
+            assert(!assignment.isDecision, "Expected assignment to be a derivation.")
```

Comment:
Shouldn't this assert that the assignment is a decision?

Reply:
There can't (and shouldn't) be any decision for the term when this is called. I believe this to be correct.

Reply:
ah, correct.